### PR TITLE
New Lucas just Dropped

### PR DESCRIPTION
**Pull request by HisashiASM**
- Added Offense Up
    - Windbox on OU Charge on frame 14
    - Offense Up charges for 2 seconds - Retains charge when hit out or special button is released
    - Offense Up is lost when whiffing Smashes, Death, Reset or OU Burst
    - Added OU Burst and OU Burst Freeze
    - Added new Smashes for Offense Up
         - USmash is OU Explosion, a HUGE explosion that deals 24% damage to a player, yet may be harder to hit floored opponents
         - Side Smash has an increased tipper and deals more damage
         - Down smash has a larger particle explosion, making it easier to get DACDS and is a little stronger
     - SFX and VFX Changes to a load of moves to fit the new OU stuff
     - DTaunt then press and hold shield on Training to quickly enable OU
   
- Added MagPulling
    - After jumping once and pressing sspecial, press magnet to gain a small speed boost going in the direction you magpulled, it is possible to b reverse if timed correctly

- Added Tiltable Side Smash
- UTaunt goes into Slip if held and now has a new animation
- SSpecial is now PK Freeze, but acts as PK Fire with more range and a smaller KB and damage
    - Now uses different voicelines
    - Shortened PK Freeze timings for output of the particle while in air
- DAir readjusted to fit new design
- Shortened DSmash hits from 3 to 1
- Shortened USmash (uncharged) times to end hitboxes on f43
- Fixed audio bug on Jab 1-2-3
- Back Air is less strong and has some of the hitbox IDs readjusted, also there is an angle change
- Up Air now comes out differently animation wise, and has a new launch angle and kb
- Forward air now has a bodybox

Needs HDR-Development/hdr-private#168 and HDR-Development/hdr-private#177

### CODE REVIEW

**Comment by Riah321** [comment]
New mechanics feel really good. Offense up seems fair in it's current state.
Down air feels really jank but probably skill issue. Super hype for this rework :)
Down taunt transitioning to trip is a little wonky in the animation.

**Comment by jobrien97** [comment]
Please edit AttackLw4's `expression` ACMD script to account for the new single-hit functionality. Controller still rumbles 3 times.

File: fighters/lucas/src/acmd/specials.rs [hunk 1]
```diff
@@ -238,59 +230,182 @@ unsafe fn lucas_special_lw_end_effect(fighter: &mut L2CAgentBase) {
...
-        EFFECT_DETACH_KIND(fighter, Hash40::new("lucas_psimagnet_end"), -1);
+        FOOT_EFFECT(fighter, Hash40::new("null"), Hash40::new("top"), 0, 0, 0, 0, 0, 0, 1, 9, 0, 9, 0, 0, 0, false);
+        FLASH(fighter, 0.01, 0.5, 1, 0.4);
     }
-    wait(lua_state, 5.0);
+    for i in 1..=50 {
+        if is_excute(fighter) {
+            if i%2==0 {
+                EFFECT_OFF_KIND(fighter, Hash40::new("lucas_pkfr_hold"), false, false);
+                EFFECT_FLW_POS(fighter, Hash40::new("lucas_pkfr_hold"), Hash40::new("top"), 0, sv_math::rand(hash40("fighter"), 4) as i32 + 12, sv_math::rand(hash40("fighter"), 4) as i32 - 2, 0, 0, 0, 0.5, true);
```
Comment (jobrien97) [inline fighters/lucas/src/acmd/specials.rs:280]: You can use the `emit` bone to make an effect spawn randomly on a character's body

**Review by jobrien97 — CHANGES_REQUESTED** [review]

File: fighters/lucas/src/opff.rs [hunk 2]
```diff
@@ -212,15 +212,170 @@ unsafe fn pk_fire_ff(boma: &mut BattleObjectModuleAccessor, stick_y: f32) {
     }
 }
 
+// Offense Up charge Handler
+pub unsafe fn offense_charge(fighter: &mut smash::lua2cpp::L2CFighterCommon, boma: &mut BattleObjectModuleAccessor, situation_kind: i32)  {
+    if(VarModule::is_flag(fighter.object(), vars::lucas::instance::SPECIAL_N_OFFENSE_UP_ACTIVE)) {
+        if fighter.is_status(*FIGHTER_STATUS_KIND_SPECIAL_N) {
+            StatusModule::change_status_request_from_script(fighter.module_accessor, *FIGHTER_LUCAS_STATUS_KIND_SPECIAL_N_FIRE, true);
+        }
```
Comment (jobrien97) [inline fighters/lucas/src/opff.rs]: This should be moved to SPECIAL_N's pre status, using set_status_kind_interrupt if the flag is on. Else, call original.
Doing this in opff causes the change_status_request_from_script to run on frame 2 of SPECIAL_N, as status_kind isn't updated until then

File: fighters/lucas/src/status.rs [hunk 3]
```diff
@@ -2,12 +2,259 @@ use super::*;
...
+        if fighter.kind() == *FIGHTER_KIND_LUCAS {
+            let charge_time = ParamModule::get_int(fighter.object(), ParamType::Agent, "attack_up_charge_time");
+            VarModule::set_int(fighter.object(), charge_time, vars::lucas::instance::SPECIAL_N_OFFENSE_UP_CHARGE_LEVEL);
+            VarModule::off_flag(fighter.object(), vars::lucas::instance::SPECIAL_N_OFFENSE_UP_ACTIVE);
+            VarModule::off_flag(fighter.object(), vars::lucas::instance::SPECIAL_N_OFFENSE_UP_INIT);
+            VarModule::off_flag(fighter.object(), vars::lucas::instance::SPECIAL_N_OFFENSE_UP_RELEASE_AFTER_WHIFF);
+        }
+    }
+}
+
```
Comment (jobrien97) [inline fighters/lucas/src/status.rs:34]: pls comment `// FIGHTER_LUCAS_STATUS_KIND_SPECIAL_N_HOLD //` here, to keep consistent with our status script organization

**Review by jobrien97 — APPROVED** [review]
lgtm :D

**Pull request merged**